plugins/modules/vmware_guest.py

```diff
@@ -3398,23 +3462,25 @@ def main():
                 set_vm_power_state(pyv.content, vm, 'poweredoff', module.params['force'])
             result = pyv.remove_vm(vm, module.params['delete_from_inventory'])
         elif module.params['state'] == 'present':
-            if module.check_mode:
-                result.update(
-                    vm_name=vm.name,
-                    changed=True,
-                    desired_operation='reconfigure_vm',
-                )
-                module.exit_json(**result)
+            # Note that check_mode is handled inside reconfigure_vm
             result = pyv.reconfigure_vm()
         elif module.params['state'] in ['poweredon', 'powered-on', 'poweredoff',
                                         'powered-off', 'restarted', 'suspended',
                                         'shutdownguest', 'shutdown-guest',
                                         'rebootguest', 'reboot-guest']:
             if module.check_mode:
+                # Identify if the power state would have changed if not in check mode
+                current_powerstate = vm.summary.runtime.powerState.lower()
+                powerstate_will_change = False
+                if ((current_powerstate == 'poweredon' and module.params['state'] not in ['poweredon', 'powered-on'])
+                        or (current_powerstate == 'poweredoff' and module.params['state'] not in ['poweredoff', 'powered-off'])
```

Comment:
should shutdown-guest be considered not changed when powered off? I think so, but documentation just says
"If state is set to shutdownguest or shutdown-guest and virtual machine exists, then the virtual machine is shutdown."

Reply:
I reviewed the module_utils set_vm_power_state function, and it does assume no change when powered off. So I updated this